src/seedu/addressbook/Main.java

```diff
@@ -107,7 +107,10 @@ private CommandResult executeCommand(Command command)  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
-            storage.save(addressBook);
+            if (!storage.save(addressBook)) {
```

Comment:
It is usually not recommended to catch one exception and throw another. Instead, if you take a look at the source code, you may see that the call to storage.save is wrapped within a try-catch block. Furthermore, the catch statement is set to catch all exceptions. This means that you can simply throw the exception from storage.save and let the original code handle it.